pandas/core/arrays/period.py

```diff
@@ -282,6 +282,10 @@ def freq(self):
         """
         return self.dtype.freq
 
+    def __array__(self, dtype=None):
+        # overriding DatetimelikeArray
+        return np.array(list(self), dtype=object)
+
```

Comment:
we should raise if dtype is not in [None, object]

Reply:
Why would we raise, instead of NumPy?

```python
In [11]: pdarr = pd.period_range('2000', periods=4).array

In [13]: np.asarray(pdser, dtype=float)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-60d3e286ef87> in <module>
----> 1 np.asarray(pdser, dtype=float)

~/sandbox/numpy/numpy/core/numeric.py in asarray(a, dtype, order)
    531
    532     """
--> 533     return array(a, dtype, copy=False, order=order)
    534
    535

TypeError: float() argument must be a string or a number, not 'Period'

In [14]: np.asarray(pdser, dtype=str)
Out[14]:
array(['2000-01-01', '2000-01-02', '2000-01-03', '2000-01-04'],
      dtype='<U64')

```

I'd rather we have less code to worry about :)

Reply:
you are ignoreing the dtype, you *must* raise if its not in a prescribe type, because you are NOT dispatching to np.nparray here